### ISSUE TITLE
Setting region to local defults to HTTPS connection to localhost

### ISSUE THREAD

**Issue by jcoetzee** [aws/aws-sdk-ruby#2792]
### Describe the bug

Using DynamoDB locally (through Docker Compose as defined in [docs](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.DownloadingAndRunning.html)). Local instance only exposes plaintext endpoint however when setting region to `local` SDK client sets endpoint to `https://localhost:8000` (i.e. TLS connection)

### Expected Behavior

Setting region to `local` should default the endpoint to plaintext connection (i.e. `http://localhost:8000`)

### Current Behavior

Endpoint is set to `https://localhost:8000`.

```
irb(main):005:0> Aws::DynamoDB::Client.new(region: 'local').list_tables
Traceback (most recent call last):
       16: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/aws-sdk-core/plugins/sign.rb:47:in `call'
       15: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/aws-sdk-core/json/error_handler.rb:10:in `call'
       14: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-dynamodb-1.79.0/lib/aws-sdk-dynamodb/plugins/crc32_validation.rb:32:in `call'
       13: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/plugins/request_callback.rb:87:in `call'
       12: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/plugins/content_length.rb:24:in `call'
       11: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/net_http/handler.rb:50:in `call'
       10: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/net_http/handler.rb:76:in `transmit'
        9: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/net_http/handler.rb:128:in `session'
        8: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/net_http/connection_pool.rb:100:in `session_for'
        7: from /Users/coetzj/.gem/ruby/2.7.0/gems/aws-sdk-core-3.168.1/lib/seahorse/client/net_http/connection_pool.rb:307:in `start_session'
        6: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/delegate.rb:83:in `method_missing'
        5: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/net/http.rb:938:in `start'
        4: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/net/http.rb:943:in `do_start'
        3: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/net/http.rb:1009:in `connect'
        2: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/net/protocol.rb:44:in `ssl_socket_connect'
        1: from /Users/coetzj/.rbenv/versions/2.7.6/lib/ruby/2.7.0/net/protocol.rb:44:in `connect_nonblock'
Seahorse::Client::NetworkingError (SSL_connect returned=1 errno=0 state=error: wrong version number)
```

### Reproduction Steps

```
irb(main):006:0> client = Aws::DynamoDB::Client.new(region: 'local')
=> #<Aws::DynamoDB::Client>
irb(main):007:0> client.config.endpoint
=> #<URI::HTTPS https://localhost:8000>
```

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

aws-sdk-dynamodb 1.79

### Environment details (Version of Ruby, OS environment)

Ruby 2.7.6, MacOS 12.6.1

**Comment by mullermp** [comment]
Thanks for reporting this. You are correct that it should be http. This bug comes from modeled endpoint rules and needs to be fixed upstream for all SDKs. As a work around, you can use an older version of dynamodb or set endpoint directly.

**Comment by alextwoods** [comment]
This has been corrected and the latest aws-sdk-dynamodb (1.80.0) and aws-sdk-dynamodbstreams (1.42.0) should correctly produce http endpoints for the `local` region.
